### PR TITLE
Next auth page changes

### PR DESCRIPTION
**Pull request by Gian-Alvarez**
I did a couple of things here. The biggest ones were adding credentials to nextauth, as well as getUser and editUser

### CODE REVIEW

File: src/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
+					return user.user;
+				} 
+				else {
+					// Returning null displays a error advising the user to check their details.
+					return null
+				}
+			}
+		}),
   ],
 })
```
Comment (xTye) [inline src/pages/api/auth/[...nextauth].ts:47]: add a line

File: src/pages/api/user/register.ts [hunk 2]
```diff
@@ -55,12 +62,11 @@ export default async function handler (
...
     // If all checks are passed.
-    
     // Hash Password.
-    
+    const hashedPass = hashPass(password);
+
     // Save New User.
-    //const userData = req.body;
-    const userData = {firstName, lastName, email, password:password};
+    const userData = {firstName, lastName, email, password:hashedPass, emailVerified:0, emailToken:0, resPassword:0, resPassToken:0};
```
Comment (xTye) [inline src/pages/api/user/register.ts:69]: Fantastic work here!

File: src/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -1,18 +1,47 @@
...
       clientId: process.env.GITHUB_ID ? process.env.GITHUB_ID : "",
       clientSecret: process.env.GITHUB_SECRET ? process.env.GITHUB_SECRET : "",
     }),
     GoogleProvider({
         clientId: process.env.GOOGLE_CLIENT_ID ? process.env.GOOGLE_CLIENT_ID : "",
         clientSecret: process.env.GOOGLE_CLIENT_SECRET ? process.env.GOOGLE_CLIENT_SECRET : "",
-      }),
-    // ...add more providers here
+    }),
+    CredentialsProvider({
```
Comment (xTye) [inline src/pages/api/auth/[...nextauth].ts:17]: Fantastic!!

Comment (xTye) [reply src/pages/api/auth/[...nextauth].ts:17]: I'll read more into this and set it up with front-end correctly

**Review by xTye — APPROVED** [review]

**Review by BraedonWatkins — APPROVED** [review]

**Pull request merged**